activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/SecurityPolicyApplicationService.java

```diff
@@ -0,0 +1,98 @@
+package org.activiti.cloud.services.core;
+
+import org.activiti.cloud.services.SecurityPolicy;
+import org.activiti.cloud.services.SecurityPolicyService;
+import org.activiti.engine.UserGroupLookupProxy;
+import org.activiti.engine.UserRoleLookupProxy;
+import org.activiti.engine.repository.ProcessDefinitionQuery;
+import org.activiti.engine.runtime.ProcessInstanceQuery;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Set;
+
+@Component
+public class SecurityPolicyApplicationService {
+
+    @Autowired(required = false)
+    private UserGroupLookupProxy userGroupLookupProxy;
+
+    @Autowired(required = false)
+    private UserRoleLookupProxy userRoleLookupProxy;
+
+    private AuthenticationWrapper authenticationWrapper = new AuthenticationWrapper();
+
+    @Autowired
+    private SecurityPolicyService securityPolicyService;
+
+
+    public ProcessDefinitionQuery processDefQuery(ProcessDefinitionQuery query, SecurityPolicy securityPolicy){
+
+        if (noSecurityPoliciesOrNoUser()){
+            return query;
+        }
+
+        Set<String> keys = definitionKeysAllowedForPolicy(securityPolicy);
+
+        if(keys != null){ //restrict query to only these keys
+            query = query.processDefinitionKeys(keys);
+        }
+        return query;
+    }
+
+    private boolean noSecurityPoliciesOrNoUser() {
+        return !securityPolicyService.policiesDefined() || authenticationWrapper.getAuthenticatedUserId()== null;
+    }
+
+    private Set<String> definitionKeysAllowedForPolicy(SecurityPolicy securityPolicy) {
+        List<String> groups = null;
+
+        if(userGroupLookupProxy!=null && authenticationWrapper.getAuthenticatedUserId()!=null){
+            groups = userGroupLookupProxy.getGroupsForCandidateUser(authenticationWrapper.getAuthenticatedUserId());
+        }
+
+        return securityPolicyService.getProcessDefinitionKeys(authenticationWrapper.getAuthenticatedUserId(),
+                groups, securityPolicy);
+    }
+
+    public ProcessInstanceQuery processInstQuery(ProcessInstanceQuery query, SecurityPolicy securityPolicy){
```

Comment:
@ryandawsonuk it might be good to rename this method to processInstanceSecuredQuery ? or something similar to make it clear that this is just wrapping a normal query + it adds the securityPolicy.

Reply:
I'll change to restrictProcessInstQuery as I think that name captures it better